fluentd/utils/sanitize_msg_chunks

```diff
@@ -0,0 +1,178 @@
+    options.dlq_dir = dir
+  end
+
+  opts.on("-r", "--require EXTRA_HANDLER", "The full path to additional handler to further evaluate",
+          "each record. The file should contain a single function with the following name and args:",
+          "  check_record(logger, time, record)") do | r |
+    options.require = r
+  end
+
+  opts.on_tail("-v", "--verbose", "Run verbosely") do |v|
+    options.log_level = Logger::DEBUG
+  end
+end.parse!
+
+$logger = Logger.new(STDOUT)
+$logger.level = options.log_level
+$logger.formatter = proc do |s, date, prog, msg|
+  "[#{s}] - #{msg}\n"
+end
+
+$logger.debug("options: #{options}")
+
+def validate_args(options)
+  unless Dir.exists?(options.dir)
+      $logger.error("Unable to continue. Buffer directory does not exist: #{options.dir}")
+      return false
+  end
+  if options.index_field.strip.empty?
+      $logger.error("The index_field can not be empty.")
+      return false
+  end
+  if options.dlq_dir != 'DROP'
+     unless options.dlq_dir.start_with?('/')
+        $logger.error("The dead letter queue must be either 'DROP' or an absolute path: #{options.dlq_dir}")
+        return false
+     end
+  end
+  require options.require unless options.require.nil?
+  return true
+end
+
+class DropDLQHandler
+
+    attr_reader :dropped
+
+    def initialize(logger)
+      @dropped = 0
+      @logger = logger
+      @logger.debug("Initializing DROP dead letter handler.  Messages will be dropped from chunk")
+    end
+
+    def handle_msg(e, time, record)
+        @dropped=+1
+        @logger.debug("Invalid record - error: '#{e.message}' time:'#{time}' record:'#{record}'")
+    end
+end
+
+class FileDLQHandler
+
+    attr_reader :dropped
+
+    def initialize(logger, dir, file)
+        @dropped = 0
+        @logger = logger
+        @file_name = File.join(dir, File.basename(file) + '.dlq')
+        unless Dir.exists?(dir)
+          logger.info ("Creating DLQ directory: #{dir}")
+          Dir.mkdir(dir)
+        end
+    end
+
+    def handle_msg(e, time, record)
+      if @file.nil?
+        @logger.info("Creating DLQ file: #{@file_name}")
+        @file = File.open(@file_name, 'w+')
+      end
+      @dropped=+1
+      @file.puts({error: "#{e.message}", time: time, record: record})
+    end
+end
+
+def init_dlq_handler(options, file)
+    return DropDLQHandler.new($logger) if options.dlq_dir == 'DROP'
+    FileDLQHandler.new($logger, options.dlq_dir, file)
+end
+
+def unpack(opts, handler, file)
+  $logger.info("Checking #{File.basename(file)}")
+  clean_file = file + '.clean'
+  p = MessagePack::Packer.new(File.open(clean_file, 'w'))
+  u = MessagePack::Unpacker.new(File.new(file, 'r'))
+  u.each do |obj|
+    begin
+      time = obj[0]
+      record = obj[1]
+      target_index = record[opts.index_field]
+      target_index = target_index.downcase
+      check_record($logger, time, record) if defined?(:check_record)
```

Comment:
I don't think this works - `if defined?(:check_record)` will always return true

Reply:
corrected.  The right way is `check_record($logger, time, record) if defined?(check_record)`